homeassistant/helpers/entity.py

```diff
@@ -1032,11 +1035,25 @@ async def async_remove(self, *, force_remove: bool = False) -> None:
         If the entity doesn't have a non disabled entry in the entity registry,
         or if force_remove=True, its state will be removed.
         """
+        if self.__remove_event is not None:
+            await self.__remove_event.wait()
+            return
+
+        self.__remove_event = asyncio.Event()
+        try:
+            await self.__async_remove_impl(force_remove)
+        finally:
+            self.__remove_event.set()
+
+    @final
+    async def __async_remove_impl(self, force_remove: bool) -> None:
```

Comment:
This could just be inlined in the try-block in `async_remove` instead.

Reply:
It's nice with a short try...except block.